hs_file_types/models/raster.py

```diff
@@ -0,0 +1,528 @@
+                                div("{% crispy form %}")
+                                with div(cls="row", style="margin-top:10px;"):
+                                    with div(cls="col-md-offset-10 col-xs-offset-6 "
+                                                 "col-md-2 col-xs-6"):
+                                        button("Save changes", type="button",
+                                               cls="btn btn-primary pull-right",
+                                               style="display: none;",
+                                               onclick="metadata_update_ajax_submit({{ "
+                                                       "form.form_id_button }}); return false;")
+                        div("{% endfor %}")
+
+        template = Template(root_div.render())
+        context_dict = dict()
+        context_dict["coverage_form"] = self.get_spatial_coverage_form()
+        context_dict["orig_coverage_form"] = self.get_original_coverage_form()
+        context_dict["cellinfo_form"] = self.get_cellinfo_form()
+        temp_cov_form = self.get_temporal_coverage_form()
+
+        update_action = "/hsapi/_internal/GeoRasterLogicalFile/{0}/{1}/{2}/update-file-metadata/"
+        create_action = "/hsapi/_internal/GeoRasterLogicalFile/{0}/{1}/add-file-metadata/"
+        if self.temporal_coverage:
+            temp_action = update_action.format(self.logical_file.id, "coverage",
+                                               self.temporal_coverage.id)
+            temp_cov_form.action = temp_action
+        else:
+            temp_action = create_action.format(self.logical_file.id, "coverage")
+            temp_cov_form.action = temp_action
+
+        context_dict["temp_form"] = temp_cov_form
+        context_dict["bandinfo_formset_forms"] = self.get_bandinfo_formset().forms
+        context = Context(context_dict)
+        rendered_html = template.render(context)
+
+        # file level form field ids need to changed so that they are different from
+        # the ids used at the resource level for the same type of metadata elements
+        rendered_html = rendered_html.replace("div_id_start", "div_id_start_filetype")
+        rendered_html = rendered_html.replace("div_id_end", "div_id_end_filetype")
+        rendered_html = rendered_html.replace("id_start", "id_start_filetype")
+        rendered_html = rendered_html.replace("id_end", "id_end_filetype")
+        for spatial_element_id in ('div_id_northlimit', 'div_id_southlimit', 'div_id_westlimit',
+                                   'div_id_eastlimit'):
+            rendered_html = rendered_html.replace(spatial_element_id,
+                                                  spatial_element_id + "_filetype", 1)
+        for spatial_element_id in ('div_id_type', 'div_id_north', 'div_id_east'):
+            rendered_html = rendered_html.replace(spatial_element_id,
+                                                  spatial_element_id + "_filetype", 1)
```

Comment:
same comment as above regarding consolidating replace() calls into a single for loop.